benchmarks/benchmarks/optimize_linprog.py

```diff
@@ -8,57 +8,30 @@
 with safe_import():
     from scipy.optimize import linprog, OptimizeWarning
 
-with safe_import() as exc:
+with safe_import():
     from scipy.optimize.tests.test_linprog import lpgen_2d, magic_square
-    from scipy.optimize._remove_redundancy import (
-        _remove_redundancy_svd,
-        _remove_redundancy_pivot_sparse,
-        _remove_redundancy_pivot_dense,
-        _remove_redundancy_id
-    )
-    from scipy.optimize._linprog_util import (
-        _presolve,
-        _clean_inputs,
-        _LPProblem
-    )
-if exc.error:
-    _remove_redundancy_svd = None
-    _remove_redundancy_pivot_sparse = None
-    _remove_redundancy_pivot_dense = None
-    _remove_redundancy_id = None
 
 with safe_import():
     from scipy.linalg import toeplitz
 
-with safe_import():
-    from scipy.sparse import csc_matrix, csr_matrix, issparse
-
-methods = [("interior-point", {"sparse": True}),
```

Comment:
If we're removing these methods from the benchmarks, should we also start a "formal" deprecation?  i.e. add a warning to `interior-point`, `revised simplex` and `simplex` to make users aware that `highs` may become the new default in a future release

Reply:
We shouldn't _deprecate_ them until HiGHS has callbacks. I'll have to take a look at the old PR that made `interior-point` the default to see what the considerations were. But in any case, i think it's ok to lose the benchmarks because we aren't retaining them for their speed; just backwards compatibility.